dds/DCPS/XTypes/TypeObject.h

```diff
@@ -1959,25 +1988,35 @@ bool operator>>(Serializer& strm, XTypes::Sequence<T>& seq)
       return false;
     }
   }
-  return true;
+  return (total_size == 0) ? true : strm.skip(end_of_seq - strm.pos());
```

Comment:
Small simplification if you want to take it.

```suggestion
  return (total_size == 0) || strm.skip(end_of_seq - strm.pos());
```